File: src/main/数组/t1018/prefixesDivBy5.go
package t1018

func prefixesDivBy5(A []int) []bool {
	n := len(A)
	res := make([]bool, n)
	num := 0
	for i := range A {
		num <<= 1
		num += A[i]
		num %= 10
		res[i] = num%5 == 0
	}
	return res
}

File: src/main/数组/t1018/PrefixesDivBy5.java
package 数组.t1018;

import java.util.ArrayList;
import java.util.List;

/**
 * @author: sotowang
 * @date: 2021/01/14 16:59
 */
public class PrefixesDivBy5 {
    public List<Boolean> prefixesDivBy5(int[] A) {
        List<Boolean> res = new ArrayList<>();
        int num = 0;
        for(int i=0;i<A.length;i++){
            num <<=1;
            num += A[i];
            num %= 10;
            res.add(num %5 == 0);
        }
        return res;
    }
}
